assets/src/components/map/controls/customControl.tsx

```diff
@@ -0,0 +1,62 @@
+import Leaflet, { ControlOptions, ControlPosition } from "leaflet"
+import { ReactNode, useEffect, useState } from "react"
+import ReactDOM from "react-dom"
+import { useMap } from "react-leaflet"
+
+const positionQuerySelector = (position: ControlPosition) => {
+  switch (position) {
+    case "topleft":
+      return ".leaflet-top.leaflet-left"
+    case "topright":
+      return ".leaflet-top.leaflet-right"
+    case "bottomleft":
+      return ".leaflet-bottom.leaflet-left"
+    case "bottomright":
+      return ".leaflet-bottom.leaflet-right"
+  }
+}
+
+export const CustomControl = ({
+  position,
+  children,
+  className,
+  insertAfterSelector,
```

Comment:
@firestack took this `insertAfterSelector` approach based on one of your suggestions in our conversation last week. I tried using `createControlComponent` to make this generic control but ultimately struggled with passing children to the generic control. I think that in order to insert this control in a particular place in the control bar we would need to continue using portal in some capacity, but please let me know if there is a different approach you see as preferable here!